tests/graph/test_graph_single_source_all_shortest_paths.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+import rustworkx
+
+
+class TestSingleSourceAllShortestPaths(unittest.TestCase):
+    def setUp(self):
+        # Set up a cycle graph with 4 nodes
+        self.cycle = rustworkx.PyGraph()
+        self.cycle_nodes = self.cycle.add_nodes_from([0, 1, 2, 3])
+        self.cycle.add_edges_from(
+            [
+                (self.cycle_nodes[0], self.cycle_nodes[1], 1),
+                (self.cycle_nodes[1], self.cycle_nodes[2], 1),
+                (self.cycle_nodes[2], self.cycle_nodes[3], 1),
+                (self.cycle_nodes[3], self.cycle_nodes[0], 1),
+            ]
+        )
+        # Set up a 4x4 grid graph
+        self.grid = rustworkx.generators.grid_graph(4, 4)
+        # Set up a disconnected graph (cycle + isolated node)
+        self.disconnected = rustworkx.PyGraph()
+        self.disconnected_nodes = self.disconnected.add_nodes_from([0, 1, 2, 3, 4])
+        self.disconnected.add_edges_from(
+            [
+                (self.disconnected_nodes[0], self.disconnected_nodes[1], 1),
+                (self.disconnected_nodes[1], self.disconnected_nodes[2], 1),
+                (self.disconnected_nodes[2], self.disconnected_nodes[3], 1),
+                (self.disconnected_nodes[3], self.disconnected_nodes[0], 1),
+            ]
+        )
+
+    def test_single_source_all_shortest_paths_cycle(self):
+        paths = rustworkx.graph_single_source_all_shortest_paths(self.cycle, self.cycle_nodes[0])
```

Comment:
Optional, but feel free to use `single_source_all_shortest_paths` directly. You can trust the dispatcher works once you added it